packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -34,19 +32,28 @@ import {
   TableDataColumnHeader,
 } from './SynapseTableRenderers'
 import { SynapseTableContext } from './SynapseTableContext'
-import { usePrefetchTableData } from './usePrefetchTableData'
+import { usePrefetchResourcesInTable } from './usePrefetchTableData'
 import { useAtomValue } from 'jotai'
-import {
-  fileIdColumnNameAtom,
-  fileVersionColumnNameAtom,
-  isLoadingNewBundleAtom,
-  tableQueryDataAtom,
-  tableQueryEntityAtom,
-} from '../QueryWrapper/QueryWrapper'
 import { isRowSelectionVisibleAtom } from '../QueryWrapper/TableRowSelectionState'
 import StyledTanStackTable from '../TanStackTable/StyledTanStackTable'
+import { useGetEntity } from '../../synapse-queries'
+
+export type SynapseTableConfiguration = Pick<
+  SynapseTableProps,
+  | 'showAccessColumn'
+  | 'showDownloadColumn'
+  | 'hideDownload'
+  | 'showDirectDownloadColumn'
+  | 'hideAddToDownloadListColumn'
+  | 'columnLinks'
+>
 
 export type SynapseTableProps = {
+  /** The row data shown in the table. */
+  rowSet: RowSet
```

Comment:
RowSet must now be passed (which is typically handled by `RowSetView`, but can be easily replaced with mock data)